libs/components/modals/src/lib/modules/modal/modal-host.component.ts

```diff
@@ -139,11 +155,38 @@ export class SkyModalHostComponent implements OnDestroy {
       injector
     );
 
+    // modal element that was just opened
+    const modalElement = modalComponentRef.location.nativeElement;
+
     modalInstance.componentInstance = modalComponentRef.instance;
 
     this.#registerModalInstance(modalInstance);
+    // hidding all elements at the modal-host level from screenreaders
+    const hostElement = this.#elRef.nativeElement;
+    let allModals = hostElement.children;
+
+    if (SkyModalHostService.openModalCount == 1) {
+      this.hideModalHostSiblings();
+      ariaPreviousValueSiblings = this.#ariaPreviousValueSiblings;
+    }
+    if (
+      SkyModalHostService.openModalCount > 1 &&
+      allModals[allModals.length - 1] == modalElement
+    ) {
+      allModals[allModals.length - 2].setAttribute('aria-hidden', true);
+    }
 
     function closeModal() {
+      // unhide siblings if last modal is closing
+      if (SkyModalHostService.openModalCount == 1) {
+        adapter.removeAriaHidden(ariaPreviousValueSiblings);
+        ariaPreviousValueSiblings.clear();
+      } else if (SkyModalHostService.topModal == hostService) {
+        // if there are more than 1 modal then unhide the one behind this one before closing it
+        allModals = hostElement.children;
+        allModals[allModals.length - 2].removeAttribute('aria-hidden');
```

Comment:
What if there are three modals and the middle once closes. Wouldn't this unhide the 1st and leave both the 1st and 3rd unhidden?

Reply:
nope! it checks that the top modal is the one that is closing and only then unhides any modals

Reply:
there is a unit test for this scenario also just to make sure no changes break that expectation https://github.com/blackbaud/skyux/blob/modal-navigation-accessibility/libs/components/modals/src/lib/modules/modal/modal.service.spec.ts#L239